InteractionBase/src/commands/CDiff.cpp

```diff
@@ -60,195 +62,15 @@ CommandResult* CDiff::executeNamed(Visualization::Item* /*source*/, Visualizatio
-											overlayGroup->addOverlay(makeOverlay(new Visualization::SelectionOverlay{item,
-												Visualization::SelectionOverlay::itemStyles().get("insert")}));
-									}
-									else
-										overlayGroup->addOverlay(makeOverlay(new Visualization::SelectionOverlay{item,
-											Visualization::SelectionOverlay::itemStyles().get("insert")}));
-								}
-								else
-									overlayGroup->addOverlay(makeOverlay(new Visualization::SelectionOverlay{item,
-										Visualization::SelectionOverlay::itemStyles().get("insert")}));
-							}
-							break;
-
-						case ChangeType::Deletion:
-							node = const_cast<Model::Node*>(revisionManager->nodeIdMap().node(id));
-							parent = node->parent();
-							if (auto item = revisionRoot->findVisualizationOf(node))
-							{
-								if (parent)
-								{
-									parentID = revisionManager->nodeIdMap().id(parent);
-									iter = changes.find(parentID);
-									if (iter != changes.end())
-									{
-										auto parentChange = iter.value();
-										if (parentChange->type() != ChangeType::Deletion)
-											overlayGroup->addOverlay(makeOverlay(new Visualization::SelectionOverlay{item,
-												Visualization::SelectionOverlay::itemStyles().get("delete")}));
-									}
-									else
-										overlayGroup->addOverlay(makeOverlay(new Visualization::SelectionOverlay{item,
-											Visualization::SelectionOverlay::itemStyles().get("delete")}));
-								}
-								else
-									overlayGroup->addOverlay(makeOverlay(new Visualization::SelectionOverlay{item,
-										Visualization::SelectionOverlay::itemStyles().get("delete")}));
-							}
-							break;
-
-						case ChangeType::Move:
-							node = const_cast<Model::Node*>(headManager->nodeIdMap().node(id));
-							if (auto item = headRoot->findVisualizationOf(node))
-								overlayGroup->addOverlay(makeOverlay(new Visualization::SelectionOverlay{item,
-									Visualization::SelectionOverlay::itemStyles().get("move")}));
-
-							node = const_cast<Model::Node*>(revisionManager->nodeIdMap().node(id));
-							if (auto item = revisionRoot->findVisualizationOf(node))
-								overlayGroup->addOverlay(makeOverlay(new Visualization::SelectionOverlay{item,
-									Visualization::SelectionOverlay::itemStyles().get("move")}));
-							break;
-
-						default:
-							break;
-					} // end switch
-
-					// test update flag
-					node = const_cast<Model::Node*>(headManager->nodeIdMap().node(id));
-					if (auto item = headRoot->findVisualizationOf(node))
-						if (change->flags().testFlag(ChangeDescription::UpdateType::Value))
-							overlayGroup->addOverlay(makeOverlay(new Visualization::SelectionOverlay{item,
-								Visualization::SelectionOverlay::itemStyles().get("update")}));
-
-
-				}
-				else
-					Q_ASSERT(false);
-			}
-
-
-	} } );
+	// TODO don't use fixed versions, removed param "name" contains selected version
+	VersionControlUI::DiffManager diffManager{"HEAD^^^^", "HEAD", managerName};
```

Comment:
Hmm, what happens to the DiffManager at the end of this method? It will be destroyed. We should probably make that class a singleton or something. Think about a good design.